docs/adoption.md

```diff
@@ -87,13 +87,16 @@ Once you have added a remote backend to your Terraform and created a service pri
 
   1. Add the following secrets as your github repository secrets, these are only available to you if you have access to create a service principal so ensure to request these if the service principal is being created for you, (Note: navigate to your Service principal under Active Directory → App registrations → select your app registration and navigate to overview):
 
-      | Secret name              | Value                                                                                                         |
-      | ------------------------ | ------------------------------------------------------------------------------------------------------------- |
-      | `AZURE_AD_CLIENT_SECRET` | This is the client secret value that was generated for the service principal in section 4 of Create a service |
-      | `AZURE_AD_CLIENT_ID`     | This is the Application (client) ID                                                                           |
-      | `AZURE_AD_TENANT_ID`     | This is the Directory (tenant) ID                                                                             |
-      | `AZURE_SUBSCRIPTION_ID`  | Navigate to subscriptions and select the Subscription ID for your subscription                                |
-      | `STATIC_SITE_NAME`       | The name of your static site                                                                                  |
+      | Secret name              | Value                                                                                                          |
+      | ------------------------ | ---------------------------------------------------------------------------------------------------------------|
+      | `AZURE_AD_CLIENT_SECRET` | This is the client secret value that was generated for the service principal in section 4 of Create a service. |
+      | `AZURE_AD_CLIENT_ID`     | This is the Application (client) ID                                                                            |
+      | `AZURE_AD_TENANT_ID`     | This is the Directory (tenant) ID                                                                              |
+      | `AZURE_SUBSCRIPTION_ID`  | Navigate to subscriptions and select the Subscription ID for your subscription                                 |
+      | `STATIC_SITE_NAME`       | The name of your static site                                                                                   |
+      | `CUSTOM_DOMAIN_NAME`     | The custom domain name you want to attach to your static site.                                                 |
```

Comment:
Is this mandatory?

What would happen if this wasn't set?
I'm think of a scenario where a LA could start adoption, but may not have a custom domain setup yet.
This [PR](https://github.com/Newarkandsherwood/housing-repairs-online-frontend/pull/51/files) is the related change.

Reply:
🤔 Not sure what would happen if the value was not given, but perhaps we could tell them to comment out the custom domain resource block until they are ready?
I'm not sure what the process for adoption is, e.g. do we **not** want them to modify the code at all?
I'm assuming it's okay to encourage them to update the code/terraform files to suit them. And maybe that could be the guidance?

Reply:
Since this was added as a secret, we should mention it in the documentation, but maybe the adoption page isn't the best place?

Reply:
Ideally adopters wouldn't need to make code changes.

We should understand would happens if this value is blank/empty (adding an empty secret) and update the documentation accordingly.

Reply:
I see, but I think the issue here is that terraform would require the domain name value, so it can't be empty; otherwise, terraform will fail at the plan & happy stages. Also, custom domain names would go through a validation step with a DNS server, so I'm not sure if using a placeholder value would work either.

Reply:
It feels like the terraform resource would be the optional bit, but maybe there's a way to conditional create the resource if a domain name value is given?

Reply:
There is a way to conditionally add a resource using the 'count' pseudo parameter in terraform.
I think this would work: 

```
resource "azurerm_static_site_custom_domain" "hrostaticwebapp" {
  count = var.custom_domain_name == "" ? 0 : 1
...
```
ref: [https://www.bitslovers.com/terraform-count/](url)